FirebaseStorage/Sources/Internal/StorageFetcherService.swift

```diff
@@ -84,4 +80,22 @@ actor StorageFetcherService {
     }
     response(shouldRetry)
   }
+
+  /// Map of apps to a dictionary of buckets to GTMSessionFetcherService.
+  private actor FetcherServiceMap {
+    static let shared = FetcherServiceMap()
```

Comment:
I think it may make sense to remove this actor and move it's contents into `StorageFetcherService`. The reason being that `StorageFetcherService` won't need to suspect for the gets and sets because the getter and setter (which can both be private) exist within the one single actor.

https://www.avanderlee.com/swift/actors/#preventing-unneeded-suspensions

WDYT?

Reply:
If we did this, I think the singleton could bubble up to `StorageFetcherService` as well.

Reply:
I finally see it. Thanks for being persistent. A bunch of unit tests need updating to make that happen and I'll follow up on it.

Reply:
See [7b15f30](https://github.com/firebase/firebase-ios-sdk/pull/13446/commits/7b15f30df4ec0f855ad5fa8a6836961517bfbbeb)

Reply:
An interesting, related point from https://www.swift.org/migration/documentation/swift-6-concurrency-migration-guide/dataracesafety#Suspension-Points

> Potential suspension points are marked in source code with the await keyword. Its presence indicates that the call might suspend at runtime, but await does not force a suspension. The function being called might suspend only under certain dynamic conditions. It’s possible that a call marked with await will not actually suspend.